src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -146,7 +149,7 @@ public function loadChoicesForValues(array $values, $value = null)
 
         // Optimize performance in case we have an object loader and
         // a single-field identifier
-        $optimize = null === $value || is_array($value) && $value[0] === $this->idReader;
+        $optimize = is_array($value) && $this->idReader && $value[0];
```

Comment:
Ok my bad, I guess I read too much, until now I read at `is_array($value) && $this->idReader === $value[0];`... 

I'm gonna take a nap :palm_tree: